US15 session visualization (base design)

saeie07 opened the pull request:
Created the base design
- Front page (Landing page with Start Button (not activated yet))
- Home (Session Design - basic buildup)
- Petal and Flower Component
- Navigation Component (will be deleted in future)
- Content from original landinpage is now within about

danielamormocea commented on client/src/App.vue:
```diff
@@ -1,87 +1,11 @@
 <script setup lang="ts">
 import { RouterLink, RouterView } from 'vue-router'
```
`RouterLink` is not used.

saeie07 replied:
adapted

danielamormocea commented on client/src/assets/base.css:
```diff
@@ -1,86 +1,66 @@
-  --vt-c-text-light-2: rgba(60, 60, 60, 0.66);
-  --vt-c-text-dark-1: var(--vt-c-white);
-  --vt-c-text-dark-2: rgba(235, 235, 235, 0.64);
+  --Font-color: #FFFFFF;
```
Should we keep them all lower cased, for consistency?

saeie07 replied:
should all be upper case. I haven't seen any lower case. or where do you mean?

danielamormocea replied:
--Font-color - exactly at the line I commented

saeie07 replied:
adapted

danielamormocea commented on client/src/components/Petal.vue:
```diff
@@ -0,0 +1,68 @@
+          R = (num >> 16) + amt,
+          G = (num >> 8 & 0x00FF) + amt,
+          B = (num & 0x0000FF) + amt;
+      return `#${(0x1000000 + (R < 255 ? R < 1 ? 0 : R : 255) * 0x10000 + (G < 255 ? G < 1 ? 0 : G : 255) * 0x100 + (B < 255 ? B < 1 ? 0 : B : 255)).toString(16).slice(1)}`;
```
This is very hard to read. Maybe add some comments to explain what this is doing?

danielamormocea commented on client/src/components/Petal.vue:
```diff
@@ -0,0 +1,68 @@
+    },
+  },
+  methods: {
+    lightenColor(color, percent) {
```
This is very hard to read. Maybe add some comments to explain what this is doing?

saeie07 replied:
will keep as is now because im still working on it and it might change.

danielamormocea replied:
I did not request a change here, just a comment would be nice for explainability :) thanks

saeie07 replied:
again, this was part of the PR in the current state and works well for this state. However, it will be changed. I am working on it today and will be different either way, I will make sure to add comments within the new PR, so I would not add additional work as of now because it will change within the next few hours.

danielamormocea commented on client/src/components/StartBlendButton.vue:
```diff
@@ -0,0 +1,49 @@
+}
+@media (max-height: 450px) {
+  .start-blend {
+    width: calc(40px + (200 * (100vh - 100px) / 350));
```
Is this necessary? Seems overly complex.

saeie07 replied:
yes, need it for a responsive display. otherwise the size does not change properly. plus still needs to be fine-tuned, so would leave it for now.

maettuu replied:
CSS is complex ^^ this is totally normal for responsiveness and height calculations

saeie07 replied:
would leave as is

danielamormocea commented on client/src/views/LandingView.vue:
```diff
@@ -43,65 +47,18 @@ const authorize = function () {
+      </nav>
+    </header>
+    <div class="container">
+      <start-blend-button />
```
Shouldn't this page have the login buttons? I'm not sure I understand why they are in the AboutView?

saeie07 replied:
See description of PR
- Content from original landinpage is now within about

login will be added once were ready / it makes sense - would just cause a lot of logins to work on design - so I would add this at the end, now we can easily jump back and forth to check the layout

danielamormocea commented on client/src/components/Navigation.vue:
```diff
@@ -0,0 +1,38 @@
+</script>
+
+<template>
+  <nav class="navigation-bar">
```
Do we actually need a Navigation component? It seems to be that the only 2 visible pages are Landing  and Home. Websocket seems for internal testing, and not sure what About is for (we could also just have a button for that if necessary).

saeie07 replied:
See description of PR
- Navigation Component (will be deleted in future)

danielamormocea replied:
Then we should not add dead code to the main, I would suggest to remove it

saeie07 replied:
see comment in WhatsApp chat

danielamormocea commented on client/src/assets/main.css:
```diff
@@ -1,35 +1,153 @@
-    display: flex;
-    place-items: center;
+/* Screen Type 1: Single Layout with Logo and Navigation */
+.type1 {
```
Maybe a more verbose name makes more sense? Maybe: "single-layout" and "three-part-layout"?

saeie07 replied:
would keep for now, since the css code is also named type 1 and type 2.

maettuu replied:
is this something we have to choose between? so either type 1 or 2?

saeie07 replied:
yes, exactly.

danielamormocea commented on client/src/router/index.ts:
```diff
@@ -16,7 +16,6 @@ const router = createRouter({
       path: '/home',
       name: 'home',
       component: HomeView,
-      meta: { requiresAuth: true }
```
Why do we remove the need for authentication here? Shouldn't only logged in users have access?

saeie07 replied:
yes, but need to be able to get into the Home Screen for design purpose without having to authenticate right now.

danielamormocea replied:
Then please do that locally - no need to push this to main

maettuu replied:
would comment it out for now :) but would not remove it

saeie07 replied:
adjusted

danielamormocea reviewed (changes requested):
Thanks a lot for this! Left a few comments, let me know what you think.

saeie07 commented:
Thanks a lot for the review. I have made a few changes to the code based on review. Most I have however, left for now, since they will be addressed in the future (navigation, authentication, etc).

danielamormocea commented on client/src/components/Flower.vue:
```diff
@@ -0,0 +1,62 @@
+<script>
+import Petal from './Petal.vue';
+
+export default {
```
@saeie07 this has not been addressed yet - It shouldn't be merged like this, without having consistency throughout the code

saeie07 replied:
adapted

danielamormocea reviewed (changes requested):
I still think we shouldn't add the code as is to main, there are still changes that could be adapted now, or else I fear they may be forgotten or left on somebody else to fix

saeie07 commented:
1. vue3 is now implemented everywhere 
2. font color writing is adapted
3. authentication - have added it as a hidden line so we can just activate it once we're ready
4. Navigation: I have left it because it's easy to work with for me. Also I have added the content we had within LandingVue in the About just to not delete anything. 

I think all change requests are good for now? or is there anything else ? :)

danielamormocea commented on client/src/assets/base.css:
```diff
@@ -1,86 +1,66 @@
+    height: 100%;
+    box-sizing: border-box;
+    overflow-x:hidden;
+    color: var(--Font-color);
```
This also needs to be lower case now, I think.
